lib/Ryu/Async.pm

```diff
@@ -439,7 +439,16 @@ sub udp_server {
         service  => $uri->port // 0,
         socktype => 'dgram'
     )->then(sub {
-        Future->done($server->write_handle->sockport)
+        Future->done($server->write_handle->sockport);
+    })->on_fail(sub {
+        my $err       = shift;
+        my $exception = Ryu::Exception->new(
+                type    => 'udp',
+                message => "UDP server failed to bind to port " . $uri->port // 0,
+                details => [$err]);
+        $log->errorf("%s - %s", $exception->message, $err);
+        Future->fail($exception);
```

Comment:
This line currently does nothing - or at least, nothing useful! It creates a new Future instance, marks it as failed, then discards it immediately.